lore/encoders.py

```diff
@@ -575,18 +575,31 @@ def cardinality(self):
 
 class OneHot(Base):
     """
-    Performs one hot encoding
+    Performs one hot encoding. One hot encoding is used to encode categorical variables into a series of indicator variables.
     """
-    def __init__(self, column, name=None, minimum_occurrences=None, compressed=False, **kwargs):
-        if compressed is True and minimum_occurrences is None:
-            raise ValueError('minimum_occurrences must be specified when compressed is True')
-        elif compressed is False and minimum_occurrences is not None:
-            logger.warning('minimum_occurrences has no effect when compressed is False')
+    def __init__(self, column, name=None, minimum_occurrences=None, percent_occurrences=None,
+                 drop_first=False, compressed=False, **kwargs):
+        """
+        :param minimum_occurrences: Minimum number of times that a level must occur so it gets a separate indicator variable. Levels below this threshold will be treated as having no effect. In other words, their encoding will be 0 across all levels. This is only used when compressed is True
+        :param percent_occurences: Minimum percentage  that a level must occur so it gets a separate indicator variable. Levels below this threshold will be treated as having no effect. In other words, their encoding will be 0 across all levels. This is only used when compressed is True
+        :drop_first: Drop the first level. This is useful for algorithms like linear regression. If the first level is not dropped the predictor matrix will be singular
+        :compressed: Prune rare levels using either minimum_occurrences or percent_occurences. Note that either percent_occurences or minimum_occurrences must be set for this.
+        """
+        if compressed is True and minimum_occurrences is None and percent_occurrences is None:
+            raise ValueError('minimum_occurrences or percent_occurences must be specified when compressed is True')
+        elif compressed is True and minimum_occurrences is not None and percent_occurrences is not None:
+            raise ValueError('You can only specify one of minimum_occurrences or percent_occurences')
+        elif compressed is False and (minimum_occurrences is not None or percent_occurrences is not None):
+            logger.warning('minimum_occurrences and percent_occurences have no effect when compressed is False')
+        self.percent_occurrences = percent_occurrences
         self.minimum_occurrences = minimum_occurrences
+        self.drop_first = drop_first
         self.compressed = compressed
         super(OneHot, self).__init__(column, name, **kwargs)
 
     def fit(self, data):
+        if self.percent_occurrences is not None:
+            self.minimum_occurrences = self.percent_occurrences*len(self.series(data))
```

Comment:
Stylistically, mathematical operators should have whitespace around them, I need to add a linter, but then I'd have to abide by it too.